source/common/scheduler.c

```diff
@@ -282,28 +300,67 @@ DWORD THREADCALL scheduler_waitable_thread( THREAD * thread )
 				break;
 		}
 	}
+#else
+	pollDetail.fd      = entry->waitable;
+	pollDetail.events  = POLLRDNORM;
+	pollDetail.revents = 0;
+
+	while( !terminate )
+	{
+		if( event_poll( thread->sigterm, 0 ) ) {
+			dprintf( "[SCHEDULER] scheduler_waitable_thread( 0x%08X ), signaled to terminate...", thread );
+			terminate = TRUE;
+		}
+		else if( event_poll( entry->pause, 0 ) ) {
+			dprintf( "[SCHEDULER] scheduler_waitable_thread( 0x%08X ), signaled to pause...", thread );
+			entry->running = FALSE;
+			while( !event_poll( entry->resume, 1000 ) );
+			entry->running = TRUE;
+			dprintf( "[SCHEDULER] scheduler_waitable_thread( 0x%08X ), signaled to resume...", thread );
+		}
+		else if( poll( &pollDetail, 1, 100 ) == POLLIN ) {
+			//dprintf( "[SCHEDULER] scheduler_waitable_thread( 0x%08X ), signaled on waitable...", thread );
+			entry->routine( entry->remote, entry->context, (LPVOID)thread->parameter2 );
+		}
+	}
+
+#endif
 
 	dprintf( "[SCHEDULER] leaving scheduler_waitable_thread( 0x%08X )", thread );
 	
 	// we acquire the lock for this block as we are freeing 'entry' which may be accessed 
 	// in a second call to scheduler_signal_waitable for this thread (unlikely but best practice).
+#ifndef _WIN32
+	dprintf( "[SCHEDULER] attempting to remove thread( 0x%08X )", thread );
+#endif
 	lock_acquire( schedulerThreadList->lock );
 	if( list_remove( schedulerThreadList, thread ) )
 	{
 		if( entry->destroy ) {
-			entry->destroy( entry->waitable, entry->context, thread->parameter2 );
+			entry->destroy( entry->waitable, entry->context, (LPVOID)thread->parameter2 );

```

Comment:
I think you're right. Even if it wasn't, you shouldn't need to cast to a void parameter in C - it happens implicitly. (same goes for malloc).